docs/source/intro.rst

```diff
@@ -11,4 +11,57 @@ As a programmatic interface to Ansible, the SDK lets you:
 * Marshall parameters for operations to different formats.
 * Pass data in native Python structures to and from Ansible.
 
-You can execute Ansible content with the SDK in any combination of local or remote subprocesses with or without containerization.
\ No newline at end of file
+SDK architecture
+----------------
+
+Ansible SDK runs in your Python application and takes the following inputs:
+
+Project
+    Logical collection of Ansible playbooks, credentials, and inventories.
+Credentials
+    Authentication with host systems.
+Inventory
+    List of host systems on which automation jobs run.
+
+With those inputs, Ansible SDK makes multi-threaded, asynchronous calls to perform automation runs against your project payload.
+You can execute Ansible content with the SDK in any combination of local or remote subprocesses.
+You can also use execution environments, which are container images packaged with all dependencies.
+
+Local execution
+***************
+
+Ansible SDK calls the local Ansible executor to perform automation runs.
+The local Ansible executor is the runtime on the application host that pulls execution environments, runs your automation jobs, and returns status to Ansible SDK.
+
+.. image:: images/sdk-local-executor.svg
+    :width: 750
+    :height: 1000
+    :align: center
+    :alt: This diagram illustrates how Ansible SDK takes inputs and calls the local Ansible executor to perform an automation run.
+
+Remote execution
+****************
+
+Ansible SDK calls the Automation mesh controller node to perform automation runs.
+The controller node is the central component that pulls execution environments and runs your automation jobs across remote nodes in the Automation mesh.
+The controller node also returns status to Ansible SDK at the end of each automation run.
+
+.. image:: images/sdk-remote-executor.svg
+    :width: 750
+    :height: 1000
+    :align: center
+    :alt: This diagram illustrates how Ansible SDK takes inputs and calls the controller node to perform an automation run across the mesh.
+
+Reporting and analytics
+***********************
+
+Ansible SDK can provide metrics data that gives you insights into your automation jobs.
+
+.. note::
+    Reporting and analytics functionality is planned for a future version and is not yet available.
+
+.. image:: images/sdk-reporting-analytics.svg
```

Comment:
The sdk sends analytics to be stored for later review. Does that need a diagram? :D Wouldn't it be more useful if we wait until this functionality is there so we can explain something not generic?

Reply:
No, we probably don't need a diagram for analytics. Initially this graphic was included in the other two diagrams then I broke it out.

I wasn't sure if we should totally drop it or not but, yeah, we can just comment this out for the time-being. Or I can delete it if anyone has a strong opinion about that.